native/core/src/common/bit.rs

```diff
@@ -553,8 +564,11 @@ pub struct BitReader {
 /// either byte aligned or not.
 impl BitReader {
     pub fn new(buf: Buffer, len: usize) -> Self {
-        let num_bytes = cmp::min(8, len);
-        let buffered_values = read_num_bytes_u64(num_bytes, buf.as_slice());
+        let buffered_values = if 8 > len {
+            read_num_bytes_u64(len, buf.as_slice())
+        } else {
+            read_u64(buf.as_slice())
```

Comment:
If `len > 8` then this will not read `len` bytes will it? Also `read_num_bytes_u64` appears to cover this case in an `unlikely` block (and handles it rather differently).

Reply:
If `len > 8` then, this will read 8 bytes only (u64).
`read_num_bytes_u64` has
```
    debug_assert!(size <= src.len());
    if unlikely(src.len() < 8) {
```
So when `size=8` we do not need the `if unlikely()` block

Reply:
Maybe we can explicitly make the condition `if src.len == 0` 
So 
```
   if src.len == 0 {
     read_u64(buf.as_slice())
   } else {
     read_num_bytes_u64(len, buf.as_slice())
   }
```

Reply:
When `if src.len == 0` is true, we cannot read anything if I understand correctly?